portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,22 +24,60 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+/** Class used to store comment data */
+class Comment {
+  String name;
+  String commentText;
+  
+  /**
+   * Constructor
+   * @param name Author of comment
+   * @param commentText Comment written by author
+   */
+  public Comment(String name, String commentText) {
+    this.name = name;
+    this.commentText = commentText;
+  }
+}
+
+
 /** Servlet that writes an list of messages as a response. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  /** Hard coded messages used for tesing JSON */
-  private List<String> messages;
+  /** Stores comments added */
+  private List<Comment> comments;
 
   @Override
   public void init(){
-    messages = new ArrayList<>(Arrays.asList("one", "two", "three"));
+    comments = new ArrayList<>();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, "name-input", "");
```

Comment:
To make it obvious what "name-input" and "" are, it is good to add argument comments. (Sorry I cannot find a link to this recommendation)

So it would be this: getParameter(request, /*name=*/"name-input", /*defaultValue=*/"")

These comments help the user understant what these raw strings are used for.

Reply:
See above